picocrt/machine/arm/crt0.c

```diff
@@ -151,6 +151,18 @@ _start(void)
 #define _REASON(r) #r
 #define REASON(r) _REASON(r)
 
+static void arm_fault_write_reg(const char *prefix, unsigned reg)
+{
+    char buf[9];
+    for (unsigned i = 0; i < 8; i++) {
```

Comment:
Looks like this could be done with putchar and avoid a temporary buffer; you're already building the string from left to right.

Reply:
Good point. Done.

Reply:
Looks awesome! Thanks!